### DISCUSSION TITLE
How to resolve correct ip address when host is running VPN

### DISCUSSION THREAD

**Discussion by mohkale** [qdm12/ddns-updater#510] [Q&A] [answered]
Hi,

I normally run my media-server with gluetun connected to mullvad but also occasionally run mulvad on my host os outside of any container. When setup like this ddns-updater will resolve to the VPN public address instead of my local network one. Do you have any suggestions for how to work around this? Previously I was using mullvad-exclude to run my ddns updater script but I'd really like to be able to plug it into a container daemon like this.

**Comment by CyberAustin** [comment]
If I'm understanding your problem right, when you run mullvad on your host os, ddns-updater updates your DNS record to reflect the VPN's IP address?

I would look at Docker networking. I'm not familiar with mullvad or what OS you're using, but maybe you could bind the docker container to your host's local IP address or the NIC? I'm assuming that mullvad makes a `tun0` interface or something and you can bind docker to `eth0` or whatever it is called on your OS.

**Reply by mohkale** [reply]
> If I'm understanding your problem right, when you run mullvad on your host os, ddns-updater updates your DNS record to reflect the VPN's IP address?

Yep.

> I would look at Docker networking. I'm not familiar with mullvad or what OS you're using

I'm running on arch linux.

> but maybe you could bind the docker container to your host's local IP address or the NIC?

I'd like support on this, I'm not too familiar with the core of how networking on Linux works. For reference I'm not seeing any extra network devices when mullvad is active :disappointed:.

**Reply by CyberAustin** [reply, marked as answer]
I think you need to look into what VPN technology you're using. I don't use (and am not familiar with) mullvad, but it looks like they default to to shadowsocks. Which is new for me, but it might be outside the traditional Linux networking realm. Maybe switch to an OpenVPN server or Wireguard that mullvad offers, then what I mentioned above might be a solution. Basically the end goal should be to get `ddns-updater` access to your computer's internet connection outside of mullvad. Personally, if this seems like a process you don't want to deal with, I would recommend looking at a Raspberry Pi Zero W. They are cheap, well suited for this task and consume negligible electricity. I wish I could help more, but I'd have to pay for mullvad just to begin to help troubleshoot.

**Reply by mohkale** [reply]
> I would recommend looking at a Raspberry Pi Zero W. They are cheap, well suited for this task and consume negligible electricity.

Yeah. Thanks for this suggestion. I had an old raspberry pi lying around. I planned to use it to enable wake-on-lan features for my server but I might as well move ddns resolution to it as well. As mentioned it takes up negligable electricity and can run in the background forever; allowing me to reach all the hosts in my home network from a remote location.
